x-pack/plugins/enterprise_search/server/lib/ml/get_ml_model_deployment_status.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MlGetTrainedModelsStatsRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { MlModelDeploymentStatus } from '../../../common/types/ml';
+
+import { ElasticsearchResponseError } from '../../utils/identify_exceptions';
+
+export const getMlModelDeploymentStatus = async (
+  modelName: string,
+  esClient: ElasticsearchClient
+): Promise<MlModelDeploymentStatus> => {
+  const modelRequest: MlGetTrainedModelsStatsRequest = {
+    model_id: modelName,
+  };
+  const modelStatsResponse = await esClient.ml.getTrainedModelsStats(modelRequest);
```

Comment:
Rather than calling any ml endpoints directly in es, you should be be using the functions ML shares from its setup contract. This ensures the correct user capabilities and space restrictions are checked.

Our shared functions are already being used in the enterprise search plugin e.g. [here](https://github.com/elastic/kibana/blob/f137cfbbdea6ec8d5271600456c983f0591da121/x-pack/plugins/enterprise_search/server/lib/pipelines/ml_inference/get_ml_inference_pipelines.ts#L38)

Looking through the existing code I notice that there are a few places where ml endpoints are being called directly in es. ([here](https://github.com/elastic/kibana/blob/f137cfbbdea6ec8d5271600456c983f0591da121/x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/pipeline_processors/get_ml_inference_pipeline_processors.ts#L117) and [here](https://github.com/elastic/kibana/blob/f137cfbbdea6ec8d5271600456c983f0591da121/x-pack/plugins/enterprise_search/server/lib/pipelines/create_pipeline_definitions.ts#L249))
Ideally these should also be changed, but I'm not suggestion it happens in this PR.